src/hotspot/share/oops/constantPool.cpp

```diff
@@ -2443,3 +2334,85 @@ void ConstantPool::verify_on(outputStream* st) {
     guarantee(pool_holder()->is_klass(),    "should be klass");
   }
 }
+
+void BSMAttributeEntries::deallocate_contents(ClassLoaderData* loader_data) {
+  MetadataFactory::free_array<u4>(loader_data, this->_offsets);
+  MetadataFactory::free_array<u2>(loader_data, this->_bootstrap_methods);
+  this->_offsets = nullptr;
+  this->_bootstrap_methods = nullptr;
+}
+
+void BSMAttributeEntries::copy_into(InsertionIterator& iter, int num_entries) const {
+  assert(num_entries + iter._cur_offset <= iter.insert_into->_offsets->length(), "must");
+  for (int i = 0; i < num_entries; i++) {
+    const BSMAttributeEntry* bsmae = entry(i);
```

Comment:
Nit: It's okay to use a simple name like `e` to represent `entry` - when you don't have different types of entries involved we don't need to encode the type in the variable name. EDIT: just like in `BSMAttributeEntries::InsertionIterator::reserve_new_entry`.